plugins/inputs/lotus/lotus.go

```diff
@@ -182,21 +197,35 @@ func (l *lotus) Start(acc telegraf.Accumulator) error {
 		for {
 			select {
 			case head := <-headCh:
-				go processHeader(ctx, acc, head, time.Now())
+				go processHeader(ctx, acc, head, time.Now(), nodePeerID)
 			case <-ctx.Done():
 				return
 			}
 		}
 	}
 	go processHeaders()
 
+	// process new messages
+	processMpoolUpdates := func() {
+		defer wg.Done()
+		for {
+			select {
+			case mpu := <-msgCh:
+				go processMpoolUpdate(ctx, acc, mpu, time.Now(), nodePeerID)
+			case <-ctx.Done():
+				return
+			}
+		}
+	}
+	go processMpoolUpdates()
+
 	return nil
 }
 
-func processTipset(ctx context.Context, node api.FullNode, acc telegraf.Accumulator, newTipSet *types.TipSet, receivedAt time.Time) {
+func processTipset(ctx context.Context, node api.FullNode, acc telegraf.Accumulator, newTipSet *types.TipSet, receivedAt time.Time, receivedFrom peer.ID) {
```

Comment:
Re: `ReceivedFrom` note. While the ambiguity is less an issue with usage in tipsets, I would prefer consist naming throughout.